prettier.config.cjs

```diff
@@ -5,4 +5,49 @@
 
 module.exports = {
 	...require("@fluidframework/build-common/prettier.config.cjs"),
+
+	// TODO: These overrides can be removed once this release group is updated to build-common 1.2.0.
+	overrides: [
+		{
+			files: "lerna.json",
+			options: {
+				printWidth: 50,
+				tabWidth: 2,
+			},
+		},
+		{
+			// Some JSON files are only ever used by JSON5-aware tools
+			files: ["tsconfig*.json", ".vscode/*.json"],
+			options: {
+				parser: "json5",
+				tabWidth: 2,
+				quoteProps: "preserve",
+			},
+		},
+		{
+			files: "*.json",
+			options: {
+				tabWidth: 2,
+				quoteProps: "preserve",
+			},
+		},
+		{
+			// YAML formatting should not use tabs, and use a 2-space indent instead
+			files: ["*.yaml", "*.yml"],
+			options: {
+				tabWidth: 2,
+				useTabs: false,
+				quoteProps: "preserve",
+			},
+		},
+		{
+			// Line break parsing in GitHub and elsewhere is inconsistent. Sometimes line breaks result in a `<br/>`, and
+			// other times they follow the CommonMark spec and ignore single line breaks. Setting proseWrap=never will
+			// condense all "markdown paragraphs" to a single line.
+			files: ["*.md", "*.mdown", "*.markdown"],
+			options: {
+				proseWrap: "never",
+			},
```

Comment:
An example of "soft line breaks" in GitHub PRs from https://github.com/tylerbutler/FluidFramework/pull/18

![image](https://github.com/microsoft/FluidFramework/assets/19589/2bf88539-edd7-4441-8d47-ea95db9bd2fc)

Reply:
Vs. proper common mark behavior in rendered markdown files:

![image](https://github.com/microsoft/FluidFramework/assets/19589/e1cdf73a-f72d-4881-9b24-691b42283b7a)

Given that we often copy/paste stuff from the repo into issues and PRs, I see badly formatted stuff fairly often.

I don't know what GitHub's rationale is. It's annoying.

Reply:
I agree I'd like Github to fix that... but to me "copy-pasting markdown to a target outside the repo does weird things in the target" still doesn't feel like a strong argument for making the experience inside the repo worse (again, IMO).

Reply:
I'll remove this so we can get the other changes in. We can continue to discuss markdown formatting. I don't think prettier is ultimately what we will want for markdown, because it's quite limited in terms of what it cleans up.